### ISSUE TITLE
flag to exit cmd2/stop the loop on command failure

### ISSUE THREAD

**Issue by teto** [python-cmd2/cmd2#688]
Follow up of https://github.com/python-cmd2/cmd2/issues/684.

This is some - late (sorry) - insight on https://github.com/python-cmd2/cmd2/pull/685. I've been trying to leverage cmd2 to help me with testing so here is my process: I have 2 scripts.
generate_transcripts et run_tests. run_tests fails as soon as a test fail, perfect ! 
Similarly, I would like `generate_transcripts` to exit when it fails to generate a transcript.

```
# Make it fail if any call fails
set -o errexit
function gen_transcript()
{
	suffix="$1"
	mptcpanalyzer "load tests/script_${suffix}.txt -t tests/trans_${suffix}.txt" "quit"
}

gen_transcript "tcp"
gen_transcript "mptcp"
...
```
E.g., I would like to have my program to stop with a non 0 exit code when "load script.txt -t text.txt" fails.
While the PR https://github.com/python-cmd2/cmd2/pull/685 targeted one command, I feel like there is the possibility to abstract the mechanism to apply it more generally. Like a flag `--exit-on-error`.
I don't know about the API, if it should be added per-command:
```
    @with_argparser(parser)
    @exit_on_error
    def do_mycommand
```
or as a global flag:
`my_cmd2_program --exit-on-error "load test.txt -t test.txt"`
What do you think ?

**Comment by tleonhardt** [comment]
@kmvanbrunt I think this request relates to some of the changes you were thinking of making.

**Comment by kmvanbrunt** [comment]
This should be fixed by #687. Neither `_generate_transcript()` or `_test_transcript()` stop when `onecmd_plus_hooks()` says to.

**Comment by teto** [comment]
@kmvanbrunt how does it translate as a cmd2 API user ? does generating a transcript always exit cmd2 ? or is it an option ? should I override onecmd_plus_hooks ?

**Comment by kmvanbrunt** [comment]
Generating a transcript will only exit the command loop if one of the commands it runs returns **True** for its **Stop** value. For instance, running **quit** while generating a transcript will stop both transcript generation and **cmd2's** command loop.

**Comment by kmvanbrunt** [comment]
Generating a transcript will stop early if any of the commands it is running returns **True** for **stop**. So as long as the command that fails sets **exit_code** and returns **True**, then you should get what you need when the app exits. 

This line of code you use to run tests should work with no changes:
```Python
mptcpanalyzer "load tests/script_${suffix}.txt -t tests/trans_${suffix}.txt" "quit"
```

**Comment by kmvanbrunt** [comment]
Closed by #696
